core/src/main/java/com/dtolabs/rundeck/core/common/PluginControlServiceImpl.java

```diff
@@ -15,16 +16,21 @@ public class PluginControlServiceImpl implements PluginControlService {
     private HashSet<String> disabledPlugins;
     private final IFramework framework;
     private final String project;
+    private final PluginRegistry pluginRegistry;
```

Comment:
TODO: If we actually take this approach, we would need to go through the rest of the methods and apply registry-level blocking.